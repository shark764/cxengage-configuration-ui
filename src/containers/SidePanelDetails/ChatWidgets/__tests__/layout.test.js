/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import ChatWidgetsDetailsPanel from '../layout';

describe('<ChatWidgetsDetailsPanel />', () => {
  let customItem;
  beforeEach(() => {
    customItem = {
      name: 'mockName'
    };
  });
  it('renders chatWidgets detailsPanel', () => {
    const rendered = shallow(
      <ChatWidgetsDetailsPanel
        id="details-panel-id"
        className="details-panel"
        userHasUpdatePermission={true}
        children={'Mock Child'}
        item={customItem}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders chatWidgets detailsPanel with no update permision', () => {
    const rendered = shallow(
      <ChatWidgetsDetailsPanel
        id="details-panel-id"
        className="details-panel"
        userHasUpdatePermission={false}
        children={'Mock Child'}
        item={customItem}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
