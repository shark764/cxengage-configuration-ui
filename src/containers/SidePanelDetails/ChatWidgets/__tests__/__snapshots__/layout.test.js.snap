// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatWidgetsDetailsPanel /> renders chatWidgets detailsPanel 1`] = `
ShallowWrapper {
  "length": 1,
}
`;

exports[`<ChatWidgetsDetailsPanel /> renders chatWidgets detailsPanel with no update permision 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
